04_linked_list/lib/linked_list.js

```diff
@@ -111,17 +111,35 @@ class LinkedList{
     this.length++
   }
 
+  kFromEnd(number) {
+    if(this.head === null){
+      throw new Error('list is empty')     
+    }
+    if (number > this.length - 1) {
+      throw new Error('number is out of range of list')
+    };
+    this.current = this.head;
+    for (let i = 0; i < this.length - 1 - number; i++){
+      this.current = this.current.next;
+    };
+    return this.current.value;
+  }
+
   //serialize method - turns Linked List into raw data -  - BigO = O(1)
-  serialize(){
-    let serializedList = JSON.stringify(this.head);
+  serialize() {
+    let serializedList = JSON.stringify(this);
     return serializedList = Buffer.alloc(serializedList.length, serializedList);
   }
 
   //de-serialize method - turns serialized linked list back into linked list  - BigO = O(1)
-  deSerialize(){
-    // console.log(this);
-    let deSerializedList = this.toString();
-    return deSerializedList = JSON.parse(deSerializedList);
+  static deSerialize(serializedList) {
+    let deSerializedList = serializedList.toString();
+    deSerializedList = JSON.parse(deSerializedList);
+    let newLinkedList = new LinkedList();

```

Comment:
This variable could also be called `result`. (You'll see this sometimes in production code.)